internal/bus/message_pipe.go

```diff
@@ -116,13 +116,9 @@ func (p *MessagePipe) Run(ctx context.Context) {
 				r.Close(ctx)
 			}
 
-			close(p.messageChannel)
-
```

Comment:
Why are we no longer closing the channel?

Reply:
It caused a panic when the agent is exiting. 
The process function was trying to send a message to a channel that was closed.
There is no need to close the channel anyway since the only time it should be closed when when the agent is being stopped.